### ISSUE TITLE
Difference between RBA and light MMM

### ISSUE THREAD

**Issue by elisakrammerfiverr** [google/lightweight_mmm#5]
Hi, 
I am trying to compare the RBA and light MMM on my data and I don't manage to understand the difference between the 2 models. Both models are regressions right ? Where does the MCMC is used in the MMM model ? @pabloduque0  @cahlheim

**Comment by pabloduque0** [comment]
Hello @elisakrammerfiverr ! 

Yes, LMMM is a regression model!

In MMM one will model sales based on media investment, so your features will be your investment/impression/clicks on your different media channels (one column per channel in a simple use case), and your target will be your sales. Not sure if you are familiar with MMM, if not feel free to [check out some online resources](https://en.wikipedia.org/wiki/Marketing_mix_modeling).

MCMC is used for estimating the parameters of the model. If you are asking where specifically in the code, you can see how we make use of Numpyro MCMC in [this line](https://github.com/google/lightweight_mmm/blob/97914cf6eab7683f13b92842477a456c7bf50307/lightweight_mmm/lightweight_mmm.py#L172).

As for RBA I am not very familiar with it so I cant really comment.

Let us know if there are any questions left unanswered!

**Comment by elisakrammerfiverr** [comment]
> Hello @elisakrammerfiverr !
> 
> Yes, LMMM is a regression model!
> 
> In MMM one will model sales based on media investment, so your features will be your investment/impression/clicks on your different media channels (one column per channel in a simple use case), and your target will be your sales. Not sure if you are familiar with MMM, if not feel free to [check out some online resources](https://en.wikipedia.org/wiki/Marketing_mix_modeling).
> 
> MCMC is used for estimating the parameters of the model. If you are asking where specifically in the code, you can see how we make use of Numpyro MCMC in [this line](https://github.com/google/lightweight_mmm/blob/97914cf6eab7683f13b92842477a456c7bf50307/lightweight_mmm/lightweight_mmm.py#L172).
> 
> As for RBA I am not very familiar with it so I cant really comment.
> 
> Let us know if there are any questions left unanswered!

@pabloduque0 Thank you, I think I understood the place of MCMC in the MMM. I am trying to use your model on my data. Do you think we could meet over zoom ? I want to ask you some questions on my pipeline

**Comment by pabloduque0** [comment]
Feel free to add any questions you have here and I will do my best answering them.

What is giving you trouble? Are you missing any functionality? Is something not clear in the docs?

**Comment by elisakrammerfiverr** [comment]
Hi @pabloduque0 I think I have issues with the optimization : 
let's take an example : if my total spending accross all channels in the train is 10 000$ then I need to set budget = jnp.sum(media_data_train.mean(axis=0)) * n_time_periods and I need to choose n_time_periods such that the budget expression would be equal to 10 000$ ? How do I chose the conditions of the optimization ? can you give me an example please ?

**Comment by elisakrammerfiverr** [comment]
@pabloduque0 also another question : why do we need budget optmization after we estimate the parameters of the regression ? I was thinking that the budget was determined by the coefficients of the regression. Where do you use the coefficients of the regression in the budget optmization ?

**Comment by pabloduque0** [comment]
Hello @elisakrammerfiverr !

Good questions, let me try to cover them.

- Optimization is done over a number of time periods, if your model has weekly data, those will be weeks, if daily, days. When you run the optimization you need to decide for how many weeks/days you want to run it. 
- The budget decision also goes related to n_time_priods. What `jnp.sum(media_data_train.mean(axis=0)) * n_time_periods` is doing is taking the average week/day and then multiplying by the number of n_time_periods so your budget will essencially be an average spending in the historic data. But this only applies if you media unit is cost/investment ($$), if you are using impressions or clicks you will need to either have a price for it, or set it manually.
- The optimization is media optimization (not model optimization). That is given the model we will use it to optimize the budget (media allocation) to maximaze sales (or other KPI). The coefficients are used in the media optimization to run predictions on the model.

Essentially for the optimization you can use an average spending on any week/days, times the number of weeks/days you are optimizing for. But you can also increase/descrease the budget depending on your goal.

- What is your media unit? (the values in your matrix what do they represent) This will allow me to give you a more concrete example.

**Comment by pabloduque0** [comment]
Closing due to lack of activity but feel free to open if there are any remaining questions :)
